### PR TITLE
Documentation for distributed CI tests pipeline

### PR DESCRIPTION
**Pull request by baijumeswani**
**Description**: 
Adding documentation for the distributed orttraining CI pipeline so that developers can leverage the pipeline by adding more tests.

### CODE REVIEW

**Review by thiagocrepaldi — COMMENTED** [review]
If you could explain with simple dummy example that works, it would help users copy and paste it.

File: tools/ci_build/github/azure-pipelines/orttraining-linux-gpu-distributed-test-ci-pipeline.yml [hunk 1]
```diff
@@ -25,7 +25,16 @@ jobs:
         -m
       DisplayName: 'Build'
 
-    # all distributed tests
+    # Entry point for all distributed CI tests.
```
Comment (edgchen1) [inline tools/ci_build/github/azure-pipelines/orttraining-linux-gpu-distributed-test-ci-pipeline.yml:28]: good info. i wonder if it's a little hard to find though. consider putting this somewhere more obvious, e.g., a readme in orttraining/orttraining/test/python.

Comment (baijumeswani) [reply tools/ci_build/github/azure-pipelines/orttraining-linux-gpu-distributed-test-ci-pipeline.yml:28]: I added a md file with more detailed description.

**Review by edgchen1 — APPROVED** [review]

**Review by thiagocrepaldi — APPROVED** [review]
That is exactly what we need

**Comment by baijumeswani** [comment]
Thank you.

**Pull request merged**